$OpenBSD: patch-src_plugins_compilergcc_compilergcc_cpp,v 1.2 2010/06/03 16:09:03 ajacoutot Exp $

Don't look for cb_console_runner in WRKINST, always assume it is there and
installed under PREFIX.

--- src/plugins/compilergcc/compilergcc.cpp.orig	Sat May 22 12:11:22 2010
+++ src/plugins/compilergcc/compilergcc.cpp	Tue Jun  1 19:35:57 2010
@@ -166,7 +166,7 @@ namespace
 {
     PluginRegistrant<CompilerGCC> reg(_T("Compiler"));
 
-    static const wxString strCONSOLE_RUNNER(platform::windows ? _T("cb_console_runner.exe") : _T("cb_console_runner"));
+    static const wxString strCONSOLE_RUNNER(platform::windows ? _T("cb_console_runner.exe") : _T("${PREFIX}/bin/cb_console_runner"));
     static const wxString strSLASH(_T("/"));
     static const wxString strSPACE(_T(" "));
     static const wxString strQUOTE(platform::windows ? _T("\"") : _T("'"));
@@ -1672,8 +1672,7 @@ int CompilerGCC::RunSingleFile(const wxString& filenam
     }
     wxString baseDir = ConfigManager::GetExecutableFolder();
 
-    if (wxFileExists(baseDir + strSLASH + strCONSOLE_RUNNER))
-        cmd << baseDir << strSLASH << strCONSOLE_RUNNER << strSPACE;
+    cmd << strCONSOLE_RUNNER << strSPACE;
 
     if (!cmd.Replace(_T("$SCRIPT"), exe_filename))
         // if they didn't specify $SCRIPT, append:
@@ -1781,7 +1780,7 @@ int CompilerGCC::Run(ProjectBuildTarget* target)
                         : EscapeSpaces(m_CdRun);
     wxString crunnStr = platform::windows
                         ? strQUOTE + baseDir + strSLASH + strCONSOLE_RUNNER + strQUOTE
-                        : EscapeSpaces(baseDir + strSLASH + strCONSOLE_RUNNER);
+                        : EscapeSpaces(strCONSOLE_RUNNER);
     wxString hostapStr = platform::windows
                         ? strQUOTE + target->GetHostApplication() + strQUOTE
                         : EscapeSpaces(target->GetHostApplication());
@@ -1820,8 +1819,6 @@ int CompilerGCC::Run(ProjectBuildTarget* target)
         // should console runner be used?
         if (target->GetUseConsoleRunner())
         {
-            if (wxFileExists(baseDir + strSLASH + strCONSOLE_RUNNER))
-            {
                 command << crunnStr << strSPACE;
 
                 if (!platform::windows)
@@ -1833,7 +1830,6 @@ int CompilerGCC::Run(ProjectBuildTarget* target)
                     QuoteStringIfNeeded(strLinkerPath);
                     command << strLinkerPath << strSPACE;
                 }
-            }
         }
     }
 
